### PR TITLE
Added get_exe_path with more features

### PR DESCRIPTION
**Pull request by bdestombe**
Use previously installed anywhere. If doesn't exist, download it.

### CODE REVIEW

**Comment by bdestombe** [comment]
As I am switching a lot between different Python environments and recreating them, the implemented features greatly reduce the number of having to download the executables.

**Comment by bdestombe** [comment]
The tests failing are not related to this PR. They were also failing in the previous commit: #345

File: nlmod/util.py [hunk 1]
```diff
@@ -89,29 +94,276 @@ def get_model_dirs(model_ws):
...
+    return exe_full_path
+
+
+def get_bin_directory(
+    bindir=None,
+    exe_name="mf6",
+    download_if_not_found=True,
+    version_tag="latest",
+    repo="executables",
+    disable_version_check=False,
```
Comment (dbrakenhoff) [inline nlmod/util.py:168]: Nicer to have a single negative? `enable_version_check=True`?

File: nlmod/util.py [hunk 2]
```diff
@@ -89,29 +94,276 @@ def get_model_dirs(model_ws):
...
+    if sys.platform.startswith("win") and not exe_name.endswith(".exe"):
+        exe_name += ".exe"
+
+    # If bindir is provided
+    if bindir is not None and not disable_version_check:
+        msg = "Incompatible arguments. If bindir is provided, disable_version_check should be True."
+        raise ValueError(msg)
+
+    use_bindir = (
+        bindir is not None and exe_name is not None and Path(bindir / exe_name).exists()
```
Comment (dbrakenhoff) [inline nlmod/util.py:224]: Don't think you need to wrap `bindir/ exe_name` in another Path call, it is already a pathlib.Path I think?

File: nlmod/util.py [hunk 3]
```diff
@@ -89,29 +94,276 @@ def get_model_dirs(model_ws):
...
+    if not flopy_metadata_fp.exists():
+        return []
+
+    meta_raw = flopy_metadata_fp.read_text()
+
+    # Remove trailing characters that are not part of the JSON.
+    while meta_raw[-3:] != "}\n]":
+        meta_raw = meta_raw[:-1]
+
+    # get metadata of most all installations
```
Comment (dbrakenhoff) [inline nlmod/util.py:311]: "most all" ?

File: nlmod/util.py (hunk 1 above)
Comment (dbrakenhoff) [reply nlmod/util.py:168]: In case this is a flopy kwarg, it's fine to use that so we match flopy's logic.

Comment (bdestombe) [reply nlmod/util.py:168]: Indeed! Great suggestion!

**Pull request merged**